depositor_test.go

```diff
@@ -0,0 +1,224 @@
+package test_main
+
+import (
+	"testing"
+
+	. "github.com/bjartek/overflow"
+)
+
+func TestDepositor(t *testing.T) {
+
+	otu := NewOverflowTest(t)
+
+	t.Run("should initialize a depositor", func(t *testing.T) {
+
+		otu.setupDepositor("account", 0.0).
+			assertDepositorBalance("account", 0.0)
+
+	})
+
+	t.Run("should update Depositor balance", func(t *testing.T) {
+
+		amount := 100.0
+
+		otu.setupDepositor("account", 0.0).
+			assertDepositorBalance("account", 0.0)
+
+		otu.addFlowToDepositor("account", amount)
+
+		otu.assertDepositorBalance("account", amount)
+
+	})
+
+	t.Run("should update Depositor balance from public account", func(t *testing.T) {
+
+		amount := 100.0
+
+		otu.setupDepositor("account", 0.0).
+			assertDepositorBalance("account", 0.0)
+
+		otu.addFlowToDepositorPublic("user1", "account", amount)
+
+		otu.assertDepositorBalance("account", amount)
+
+	})
+
+	t.Run("should send ExampleNFT with setup", func(t *testing.T) {
+
+		amount := 100.0
+
+		otu.setupDepositor("account", 0.0).
+			addFlowToDepositor("account", amount).
+			assertDepositorBalance("account", amount)
+
+		otu.cleanup("user1")
+
+		otu.O.Tx(
+			"ExampleNFT/setup_account_example_nft",
+			WithSigner("user1"),
+		).
+			AssertSuccess(t)
+
+		otu.assertExampleNFTLength("user1", 0)
+
+		otu.O.Tx("ExampleNFT/try_send_example_nft_with_depositor",
+			WithSigner("account"),
+			WithArg("recipient", "user1"),
+		).
+			AssertSuccess(t).
+			AssertEvent(t, "Deposit", map[string]interface{}{
+				"to": otu.O.Address("user1"),
+			})
+
+	})
+
+	t.Run("should refund all fees when withdrawn", func(t *testing.T) {
+
+		amount := 100.0
+
+		otu.setupDepositor("account", 0.0).
+			addFlowToDepositor("account", amount).
+			assertDepositorBalance("account", amount)
+
+		otu.cleanup("user1")
+
+		event := otu.O.Tx(
+			"ExampleNFT/mint_and_deposit_with_depositor",
+			WithSigner("account"),
+			WithArg("recipient", "user1"),
+		).
+			Print().
+			AssertSuccess(t).
+			AssertEvent(t, "TicketDeposited", map[string]interface{}{
```

Comment:
Thank you for pointing that out. 
Overflow supports checking the "suffix" of the event and therefore we don't have to put the complete event name (i.e. A.xxx.ContractName.EventName) to assert the event. 
The reason I put Contract Name in front of some of the event name is because they are not unique in that transaction (e.g. ExampleNFT.Deposit and FlowToken.Deposit). 
Adding the contract name there to make sure I am asserting the correct event.

Reply:
Makes sense, appreciate the clarification 🙂